docs/dg/dev/guidelines/performance-guidelines/database-performance-quidelines.md

```diff
@@ -0,0 +1,133 @@
+---
+
+## Performance Guidance: Batch Processing Traits in Propel ORM
+
+### Prerequirements
+Update your Propel package to the: [spryker/propel:^3.45.0](https://github.com/spryker/propel/releases/tag/3.45.0)
+
+### ActiveRecordBatchProcessorTrait
+- This trait provides performance optimizations for database operations by allowing you to save Propel entities in batches.
+- It enables saving multiple records in a single operation, helping to resolve the - https://docs.spryker.com/docs/dg/dev/troubleshooting/troubleshooting-performance-issues/n+1-problem.html
+- It reduces database round trips and improves overall performance.
+
+#### Basic Usage Example:
+``` php
+class EntityManager {
+    use ActiveRecordBatchProcessorTrait;
+
+    public function processItems(array $itemTransfers): void {
+        // Load or create all entities at once
+        $items = $this->findOrCreateEntities($itemTransfers);
+
+        // Queue entities for batch processing
+        foreach ($items as $item) {
+            $this->persist($item);
+        }
+
+        // Save all collected entities in a single batch
+        $this->commit();
+    }
+}
+```
+#### Key Methods:
+- `persist($entity)`: Adds an entity to the batch queue. Handles separation for insert and update operations.
+- `commit()`: Commits all queued entities to the database in a single batch.
+
+### CascadeActiveRecordBatchProcessorTrait
+- Extends `ActiveRecordBatchProcessorTrait` by adding recursive cascade-saving functionality for related entities. 
+- Helps manage complex entity relationships during batch processing. 
+- Works with `BatchEntityPostSaveInterface`, which enables organizing `postSave` events for related entities in a batch-safe manner.
+
+#### Basic Usage Example:
+
+```php
+class AbstractSpySalesOrderItem extends BaseSpySalesOrderItem implements BatchEntityPostSaveInterface
+{
+    // properties re-declared protected to prevent their direct use from an entity
+    use CascadeActiveRecordBatchProcessorTrait {
+        persist as protected;
+        commit as protected;
+        commitIdentical as protected;
+    }
+    public function batchPostSave(): void
+    {
+        if ($this->statusChanged && $this->getIdSalesOrderItem()) {
+            /** @var \Orm\Zed\Sales\Persistence\SpySalesOrderItem $salesOrderItemEntity */
+            $salesOrderItemEntity = $this;
+            $omsOrderItemStateHistoryEntity = $this->createOmsOrderItemStateHistoryEntity();
+            $omsOrderItemStateHistoryEntity->setOrderItem($salesOrderItemEntity);
+            $omsOrderItemStateHistoryEntity->setFkOmsOrderItemState($this->getFkOmsOrderItemState());
+
+            $this->sharedPersist($omsOrderItemStateHistoryEntity);
+        }
+
+        $this->statusChanged = false;
+    }
+}
+```
+With this implementation, the following code:
+```php 
+public function updateOrCreateSalesOrderItems(array $salesOrderItemsData): vois
```

Comment:
Please mention the class. It is important to see that it is the child of AbstractSpySalesOrderItem, or is actually the AbstractSpySalesOrderItem